src/screens/Customers/CustomerProfile.tsx

```diff
@@ -36,14 +37,63 @@ const styles = (theme: Theme) =>
   });
 
 interface CustomerProps extends RouteComponentProps {
+  match: { params: string; isExact: boolean; path: string; url: string; }
   classes: { root: string; content: string; header: string; button: string; buttonSecondary: string };
+  siteName: string;
   location: any;
 }
 
 function CustomerProfile(props: CustomerProps) {
   const { classes, match } = props;
-  const customer: CustomerRecord = props.location.state.customer;
+  const customer = props.location.state.customer;
 
+  // helper functions for getting data
+  const isCurrentReading = (mr: MeterReadingRecord): boolean => {
+    const period: number = parseInt(mr.date.slice(5, 7));
+    const currPer: number = moment().month() + 1;
```

Comment:
Lets move this period calculation and all moment-related stuff to `momentUtils` or another utils file as functions so that it's only written once since it's a little confusing and can easily be messed up (especially the slice by 5 & 7, adding 1 to current period, and the mod 12 in `isStartingMeter`). I think `getCurrentPeriod` and `getPreviousPeriod` would be really helpful helper functions that add a layer of abstraction over working with moment.

By abstracting away date calculations into functions in util functions that we just call, we reduce any type of errors that occur (off-by-1 or 0-index) since calculations will then always be done in the same way.